### PR TITLE
test: adjust locales, currencies and add to cart tests a bit

### PR DESCRIPTION
**Pull request by AndreyYevtukhov**
Improve some actions related to locales, currencies and add to cart tests that should make e2es more stable 

closes: -

### CODE REVIEW

File: apps/storefront-e2e/src/support/page_fragments/header.fragment.ts [hunk 1]
```diff
@@ -34,30 +34,39 @@ export class HeaderFragment {
...
-    }).as('productOrSearchRequests');
-    // hydrate
-    this.getLocaleButton().click();
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(500);
-    this.getLocaleButton().click();
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    this.initProductsUpdateInterceptor();
+
+    this.getLocaleButton().trigger('mouseenter').wait(75).click();
```
Comment (tolerants) [inline apps/storefront-e2e/src/support/page_fragments/header.fragment.ts]: I'd probably move this `75` somewhere to utility layer, so to have a constant for ui delay.

Comment (tolerants) [inline apps/storefront-e2e/src/support/page_fragments/header.fragment.ts]: What is the trick - why we should trigger mouse enter first and wait a bit and then click on it?

**Review by tolerants — DISMISSED** [review]

File: apps/storefront-e2e/src/support/page_fragments/header.fragment.ts (hunk 1 above)
Comment (AndreyYevtukhov) [reply apps/storefront-e2e/src/support/page_fragments/header.fragment.ts]: The `mouseenter` event triggers component hydration 

The wait is needed to postpone a click on the element while `js` file is being loaded and the component is re-rendered

I've tried to solve it in a better way (without waiting) by checking the `defer-hydration` attribute existence on the element - but it didn't work. Let's have it as it is now, and when Stan is back maybe we will try to improve it further

Comment (AndreyYevtukhov) [reply apps/storefront-e2e/src/support/page_fragments/header.fragment.ts]: I'm not sure that this number is consistent and will be used somewhere else(
It's more like a temporary solution for hydration, not a final one...

File: apps/storefront-e2e/src/support/page_fragments/search.fragment.ts [hunk 2]
```diff
@@ -22,25 +22,25 @@ export class SearchFragment {
...
-      )}*`
-    ).as('searchQuery');
+    const encoded = encodeURIComponent(text).replace(/%20/g, '+');
+
+    cy.intercept(`/catalog-search-suggestions?q=${encoded}*`).as('searchQuery');
 
     // trigger hydration
-    this.getTypeahead().click();
-    // wait for hydration
+    cy.intercept('/assets/box.component-*.js').as('searchHydationFiles');
```
Comment (tolerants) [inline apps/storefront-e2e/src/support/page_fragments/search.fragment.ts:30]: But we shouldn't rely on details of how assets are bundled.... and why this component has been chosen? :)

File: apps/storefront-e2e/src/support/page_fragments/search.fragment.ts [hunk 3]
```diff
@@ -22,25 +22,25 @@ export class SearchFragment {
...
+    this.getInput().click();
+    cy.wait('@searchHydationFiles');
+
+    // wait till search elements are re-rendered
     // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(500);
-    this.getInput().type(text, { delay: 10, force: true });
+    cy.wait(250);
```
Comment (tolerants) [inline apps/storefront-e2e/src/support/page_fragments/search.fragment.ts:36]: All these magic 250, 150, 125, 75 delays are killing me :)

**Review by tolerants — CHANGES_REQUESTED** [review]

File: apps/storefront-e2e/src/support/page_fragments/search.fragment.ts (hunk 2 above)
Comment (AndreyYevtukhov) [reply apps/storefront-e2e/src/support/page_fragments/search.fragment.ts:30]: Yep, that looks weird, but that's the only thing which proves that hydration was triggered 

I've just checked which one is being loaded all the time on search hydration + this one is the last one

File: apps/storefront-e2e/src/support/page_fragments/search.fragment.ts (hunk 3 above)
Comment (AndreyYevtukhov) [reply apps/storefront-e2e/src/support/page_fragments/search.fragment.ts:36]: I hope we will be able to get rid of all of these someday

**Review by tolerants — APPROVED** [review]

**Pull request merged**